test/jdk/javax/swing/JFrame/DefaultCloseOperation.java

```diff
@@ -0,0 +1,231 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Frame;
+import java.awt.FlowLayout;
+import java.awt.Window;
+import java.awt.event.ItemEvent;
+import java.awt.event.WindowEvent;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+
+/*
+ * @test
+ * @summary test for defaultCloseOperation property for Swing JFrame and JDialog
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual DefaultCloseOperation
+ */
+
+public class DefaultCloseOperation extends JPanel {
+    private static final String INSTRUCTIONS = """
+        This is a manual test (requires user interaction) which tests the
+        "defaultCloseOperation" property for swing's JFrame and JDialog.
+        This property enables a program to tell swing to automatically
+        handle a user's "Close" operation on a JFrame or JDialog in one of
+        the following ways:
+
+         - Do nothing - do not do anything when the user closes the window
+         - Hide - automatically hide the window when the user closes it
+         - Dispose - automatically dispose the window when the user closes it
+
+        To run this test, do the following steps (if an exception is thrown
+        at anytime during the test, the test has failed, else it passed):
+
+         -  Click the "Open Frame" button (a TestFrame will appear)
+         -  On the TestFrame, select "Close" from the system menu (the window should go away)
+         -  Select "Do Nothing" from the "JFrame Default Close Operation" ComboBox
+         -  Click the "Open Frame" button
+         -  On the TestFrame, select "Close" from the system menu (the window should remain open)
+         -  Select "Dispose" from the "JFrame Default Close Operation" ComboBox
+         -  On the TestFrame, select "Close" from the system menu (the window should go away)
+
+
+         -  Click the "Open Frame" button
+         -  Click the "Open Dialog" button (a TestDialog will appear)
+         -  On the TestDialog, select "Close" from the system menu (the window should go away)
+         -  Select "Do Nothing" from the "JDialog Default Close Operation" ComboBox
+         -  Click the "Open Dialog" button
+         -  On the TestDialog, select "Close" from the system menu (the window should remain open)
+         -  Select "Dispose" from the "JDialog Default Close Operation" ComboBox
+         -  On the TestDialog, select "Close" from the system menu (the window should go away)
+        """;
```

Comment:
I guess this test should disable Pass and Fail buttons. The Fail button should remain disabled: the test will fail automatically. Yet the Pass button may always be enabled, or it could be enabled only after the tester cycles through all the scenarios.

Reply:
I think it's fine the way it is now. Just in case the user sees some unusual failure.

Reply:
Anyway I submitted [JDK-8317111](https://bugs.openjdk.org/browse/JDK-8317111): Provide access to Pass/Fail buttons of PassFailJFrame.

There could be scenarios where it's useful to control access to Pass/Fail buttons.